client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/overview/OverviewScreen.kt

```diff
@@ -1,52 +1,243 @@
 package org.thechance.common.presentation.overview
 
+import androidx.compose.foundation.Image
 import androidx.compose.foundation.background
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Divider
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.collectAsState
-import androidx.compose.runtime.getValue
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import cafe.adriel.voyager.core.screen.Screen
-import cafe.adriel.voyager.navigator.LocalNavigator
-import cafe.adriel.voyager.navigator.currentOrThrow
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import cafe.adriel.voyager.navigator.Navigator
+import cafe.adriel.voyager.navigator.tab.LocalTabNavigator
+import cafe.adriel.voyager.navigator.tab.TabNavigator
+import com.beepbeep.designSystem.ui.composable.BpOutlinedButton
 import com.beepbeep.designSystem.ui.theme.Theme
-import org.koin.core.component.KoinComponent
-import org.koin.core.component.inject
+import org.thechance.common.presentation.base.BaseScreen
+import org.thechance.common.presentation.composables.OverviewDropDown
+import org.thechance.common.presentation.main.RestaurantsTab
+import org.thechance.common.presentation.main.TaxisTab
+import org.thechance.common.presentation.main.UsersTab
+import org.thechance.common.presentation.resources.Resources
+import org.thechance.common.presentation.util.kms
 
-object OverviewScreen : Screen, KoinComponent {
+object OverviewScreen :
+    BaseScreen<OverviewScreenModel, OverviewUiEffect, OverviewUiState, OverviewInteractionListener>() {
 
-    private val screenModel: OverviewScreenModel by inject()
+    private fun readResolve(): Any = OverviewScreen
+
+    override fun onEffect(effect: OverviewUiEffect, navigator: Navigator) {
+
+    }
 
     @Composable
-    override fun Content() {
-        val navigate = LocalNavigator.currentOrThrow
-        val state by screenModel.state.collectAsState()
+    override fun OnRender(state: OverviewUiState, listener: OverviewInteractionListener) {
+
+        Column(
+            modifier = Modifier.background(Theme.colors.surface).fillMaxSize(),
+            verticalArrangement = Arrangement.spacedBy(24.kms)
+        ) {
+
+            Column(
```

Comment:
for example you can put this column in a private function because it contains many component and it represent the card on the top in the screen so we can put it in a separate function